lib/src/auth.dart

```diff
@@ -55,35 +50,55 @@ abstract class Authenticator {
     if (credentials == null) {
       return;
     }
-    String accessToken = credentials.accessToken;
-    Map<String, String> revokeAccess = new Map<String, String>();
-    revokeAccess[kTokenKey] = accessToken;
-    revokeAccess[kTokenTypeHintKey] = 'access_token';
+    List<Map> tokens = new List<Map>();
+    Map accessToken = {
+      kTokenKey: credentials.accessToken,
+      kTokenTypeHintKey: 'access_token',
+    };
+    tokens.add(accessToken);
+
+    if (credentials.refreshToken != null) {
+      Map refreshToken = {
+        kTokenKey: credentials.refreshToken,
+        kTokenTypeHintKey: 'refresh_token',
+      };
+      tokens.add(refreshToken);
+    }
 
-    // Retrieve the client ID and secret.
-    String clientId = _grant.identifier;
-    String clientSecret = _grant.secret;
+    for (int i = 0; i < tokens.length; i++) {
+      String token = tokens[i][kTokenKey];
+      String tokenType = tokens[i][kTokenTypeHintKey];
+      Map<String, String> revokeAccess = new Map<String, String>();
+      revokeAccess[kTokenKey] = token;
+      revokeAccess[kTokenTypeHintKey] = tokenType;
 
-    // TODO(bkonyi) handle cases where clientSecret isn't used.
-    String userInfo = '$clientId:$clientSecret';
+      // TODO(bkonyi) we shouldn't have hardcoded urls like this. Move to common
+      // file with all API related strings.
+      Uri path = Uri.parse(r'https://www.reddit.com/api/v1/revoke_token');
```

Comment:
Avoid LHS type unless it's unclear from context or type inference doesn't infer the desired type.